src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala

```diff
@@ -102,24 +103,30 @@ package object servlets extends DebugEnhancedLogging {
         .map(charSet => managed(new ZipInputStream(fileItem.getInputStream, Charset.forName(charSet))))
         .getOrElse(managed(new ZipInputStream(fileItem.getInputStream)))
     }
+  }
+
+  implicit class RichMultipartConfig(config: MultipartConfig) {
+    def moveNonZips(srcItems: Iterator[FileItem], targetDir: File): Try[Unit] = {
+      srcItems.toStream
+        .map(moveIfNonZip(_, targetDir))
+        .failFastOr(Success(()))
+    }
 
-    def copyNonZipTo(dir: File): Try[Unit] = {
-      if (fileItem.name.isBlank) Success(()) // skip form field without selected files
-      else if (fileItem.isZip) Failure(ZipMustBeOnlyFileException(fileItem))
-      else
-        managed(fileItem.getInputStream)
-          .apply(inputStream => Try { Files.copy(inputStream, (dir / fileItem.name).path) })
+    private def moveIfNonZip(srcItem: FileItem, targetDir: File): Try[Unit] = {
+      logger.info(s"staging upload: size=${ srcItem.size } contentType=${ srcItem.contentType } $targetDir/${ srcItem.name }")
+      if (srcItem.name.isBlank) Success(()) // skip form field without selected files
+      else if (srcItem.isZip) Failure(ZipMustBeOnlyFileException(srcItem))
+      else Try {
+        val f = UUID.randomUUID().toString
+        val location = File(config.location.getOrElse(throw ConfigurationException("multipart.location is missing")))
+        srcItem.part.write(f) // for big files in practice a rename to a known file in location
```

Comment:
I don't understand why you need to write something here. The comment isn't clear.